projects/flyout/src/modules/flyout/flyout.service.ts

```diff
@@ -86,6 +86,13 @@ export class SkyFlyoutService implements OnDestroy {
           if (event instanceof NavigationStart) {
             this.close();
           }
+
+          // Sanity check - if the host still exists after animations should have completed - remove host
+          setTimeout(() => {
```

Comment:
No. The issue is that we are doing that - but this sanity check is doing is ensuring that we remove things if for some reason that fails. 

The issue from point 1 in the description caused a state where the flyout closed but the service saw it as being in a state where the host shouldn't be removed (such as when you open a new flyout quickly and we want to just replace things). This caused the router watcher to hit a path where the host wasn't removed as it called "close" on an already closed flyout and so the `close` observable that we use to remove the host was never called because that fires on the `animation.done` and there is no animation when closing an already closed flyout.

Let me take a look and see if we can possibly use the stable stuff we are doing for the a11y stuff to make this a little less arbitrary.